File: core/src/main/java/vn/loitp/views/autosize/imagebuttonwithsize/ImageButtonWithSize.java
package vn.loitp.views.autosize.imagebuttonwithsize;

import android.content.Context;
import android.content.res.Configuration;
import android.util.AttributeSet;
import android.widget.ImageButton;

import vn.loitp.core.utilities.LLog;
import vn.loitp.utils.util.ConvertUtils;
import vn.loitp.utils.util.ScreenUtils;

/**
 * Created by LENOVO on 4/19/2018.
 */

public class ImageButtonWithSize extends ImageButton {
    private final String TAG = getClass().getSimpleName();

    public ImageButtonWithSize(Context context) {
        super(context);
    }

    public ImageButtonWithSize(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public ImageButtonWithSize(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public ImageButtonWithSize(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
    }

    private final int defaultSizePortrait = 50;
    private final int defaultSizeLandscape = 150;
    private int portraitSizeW = defaultSizePortrait;
    private int portraitSizeH = defaultSizePortrait;
    private int landscapeSizeW = defaultSizeLandscape;
    private int landscapeSizeH = defaultSizeLandscape;

    public void setPortraitSizeWInPx(int portraitSizeW) {
        this.portraitSizeW = portraitSizeW;
        updateSize();
    }

    public void setPortraitSizeHInPx(int portraitSizeH) {
        this.portraitSizeH = portraitSizeH;
        updateSize();
    }

    public void setLandscapeSizeWInPx(int landscapeSizeW) {
        this.landscapeSizeW = landscapeSizeW;
        updateSize();
    }

    public void setLandscapeSizeHInPx(int landscapeSizeH) {
        this.landscapeSizeH = landscapeSizeH;
        updateSize();
    }

    public void setPortraitSizeWInDp(float portraitSizeW) {
        this.portraitSizeW = ConvertUtils.dp2px(portraitSizeW);
        updateSize();
    }

    public void setPortraitSizeHInDp(float portraitSizeH) {
        this.portraitSizeH = ConvertUtils.dp2px(portraitSizeH);
        updateSize();
    }

    public void setLandscapeSizeWInDp(float landscapeSizeW) {
        this.landscapeSizeW = ConvertUtils.dp2px(landscapeSizeW);
        updateSize();
    }

    public void setLandscapeSizeHInDp(float landscapeSizeH) {
        this.landscapeSizeH = ConvertUtils.dp2px(landscapeSizeH);
        updateSize();
    }

    private void updateSize() {
        boolean isPortrait = ScreenUtils.isPortrait();
        LLog.INSTANCE.d(TAG, "updateSize isPortrait " + isPortrait);
        LLog.INSTANCE.d(TAG, "size portrait: " + portraitSizeW + "x" + portraitSizeH);
        LLog.INSTANCE.d(TAG, "size landscape: " + landscapeSizeW + "x" + landscapeSizeH);
        if (isPortrait) {
            this.getLayoutParams().width = portraitSizeW;
            this.getLayoutParams().height = portraitSizeH;
        } else {
            this.getLayoutParams().width = landscapeSizeW;
            this.getLayoutParams().height = landscapeSizeH;
        }
        invalidate();
    }

    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        super.onConfigurationChanged(newConfig);
        updateSize();
    }
}

File: core/src/main/java/com/loitpcore/views/autoSize/imageButtonWithSize/ImageButtonWithSize.kt
package com.loitpcore.views.autoSize.imageButtonWithSize

import android.content.Context
import android.content.res.Configuration
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatImageButton
import com.loitpcore.utils.util.ConvertUtils
import com.loitpcore.utils.util.ScreenUtils

/**
 * Created by Loitp on 04,August,2022
 * Galaxy One company,
 * Vietnam
 * +840766040293
 * freuss47@gmail.com
 */
class ImageButtonWithSize : AppCompatImageButton {
//    private val logTag = javaClass.simpleName

    constructor(context: Context) : super(context)
    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)
    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    )

    private val defaultSizePortrait = 50
    private val defaultSizeLandscape = 150
    private var portraitSizeW = defaultSizePortrait
    private var portraitSizeH = defaultSizePortrait
    private var landscapeSizeW = defaultSizeLandscape
    private var landscapeSizeH = defaultSizeLandscape

    fun setPortraitSizeWInPx(portraitSizeW: Int) {
        this.portraitSizeW = portraitSizeW
        updateSize()
    }

    fun setPortraitSizeHInPx(portraitSizeH: Int) {
        this.portraitSizeH = portraitSizeH
        updateSize()
    }

    fun setLandscapeSizeWInPx(landscapeSizeW: Int) {
        this.landscapeSizeW = landscapeSizeW
        updateSize()
    }

    fun setLandscapeSizeHInPx(landscapeSizeH: Int) {
        this.landscapeSizeH = landscapeSizeH
        updateSize()
    }

    fun setPortraitSizeWInDp(portraitSizeW: Float) {
        this.portraitSizeW = ConvertUtils.dp2px(portraitSizeW)
        updateSize()
    }

    fun setPortraitSizeHInDp(portraitSizeH: Float) {
        this.portraitSizeH = ConvertUtils.dp2px(portraitSizeH)
        updateSize()
    }

    fun setLandscapeSizeWInDp(landscapeSizeW: Float) {
        this.landscapeSizeW = ConvertUtils.dp2px(landscapeSizeW)
        updateSize()
    }

    fun setLandscapeSizeHInDp(landscapeSizeH: Float) {
        this.landscapeSizeH = ConvertUtils.dp2px(landscapeSizeH)
        updateSize()
    }

    private fun updateSize() {
        val isPortrait = ScreenUtils.isPortrait
        if (isPortrait) {
            this.layoutParams.width = portraitSizeW
            this.layoutParams.height = portraitSizeH
        } else {
            this.layoutParams.width = landscapeSizeW
            this.layoutParams.height = landscapeSizeH
        }
        invalidate()
    }

    public override fun onConfigurationChanged(newConfig: Configuration) {
        super.onConfigurationChanged(newConfig)
        updateSize()
    }
}
